configs/components/rubygem-ffi.rb

```diff
@@ -1,6 +1,6 @@
 component "rubygem-ffi" do |pkg, settings, platform|
-  pkg.version '1.9.25'
-  pkg.md5sum "e8923807b970643d9e356a65038769ac"
+  pkg.version '1.12.2'
```

Comment:
Any gem that is shared by multiple projects needs special treatment. ffi in particular is sketchy for bolt bc it was not compiling on osx. BUT i think maybe we removed the OSX that it was failing for? There are some other gems you will want to grep /projects for (I think like multi_json). 
```
cas@cas-ThinkPad-T460p:~/working_dir/puppet-runtime/configs/projects$ ag multi_json
agent-runtime-5.5.x.rb
40:  proj.component 'rubygem-multi_json'

agent-runtime-master.rb
32:  proj.component 'rubygem-multi_json'

pe-installer-runtime.rb
134:  proj.component 'rubygem-multi_json'

agent-runtime-6.4.x.rb
30:  proj.component 'rubygem-multi_json'

bolt-runtime.rb
170:  proj.component 'rubygem-multi_json'
```

Reply:
Basically if there is a gem that is shared by other runtimes (especially puppet agent) we need to either ask those maintainers if it is OK to bump or we need to manage the gem with a version like this pattern: 
https://github.com/puppetlabs/puppet-runtime/blob/41063e16d7bfb9add8cefcfd1d159cbbc792c9c1/configs/projects/bolt-runtime.rb#L10
https://github.com/puppetlabs/puppet-runtime/blob/41063e16d7bfb9add8cefcfd1d159cbbc792c9c1/configs/components/rubygem-deep_merge.rb#L3-L13